// next-route-visualizer/dist/visualizer/components/flow/flow.js
"use strict";
/* eslint-disable react-hooks/exhaustive-deps */
"use client";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireWildcard(require("react"));
var _reactflow = _interopRequireWildcard(require("reactflow"));
require("reactflow/dist/base.css");
var _nodesFlow = _interopRequireDefault(require("./nodes-flow"));
var _icons = require("../icons");
var _constants = require("../../constants");
var _flowModule = _interopRequireDefault(require("../../styles/flow.module.css"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var __jsx = _react["default"].createElement;
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var nodeTypes = {
  routeNode: _nodesFlow["default"]
};
var defaultEdgeOptions = {
  animated: true,
  type: "default"
};
var Flow = function Flow(props) {
  var _selectedNode$data4, _selectedNode$data5, _selectedNode$data6, _selectedNode$data7, _selectedNode$data8, _selectedNode$data8$n, _selectedNode$data9, _selectedNode$data9$n, _selectedNode$data10, _selectedNode$data10$, _selectedNode$data11, _selectedNode$data11$;
  var _useNodesState = (0, _reactflow.useNodesState)(props.initialNodes),
    _useNodesState2 = (0, _slicedToArray2["default"])(_useNodesState, 3),
    nodes = _useNodesState2[0],
    setNodes = _useNodesState2[1],
    onNodesChange = _useNodesState2[2];
  var _useEdgesState = (0, _reactflow.useEdgesState)(props.initialEdges),
    _useEdgesState2 = (0, _slicedToArray2["default"])(_useEdgesState, 3),
    edges = _useEdgesState2[0],
    setEdges = _useEdgesState2[1],
    onEdgesChange = _useEdgesState2[2];
  var _useState = (0, _react.useState)(),
    selectedNodeID = _useState[0],
    setSelectedNodeID = _useState[1];
  var refreshPage = function refreshPage() {
    window.location.reload();
  };
  var closePanel = function closePanel() {
    setSelectedNodeID(undefined);
  };
  var getNodeSelected = function getNodeSelected(event, node) {
    console.log(node);
    return selectedNodeID === node.id ? setSelectedNodeID(undefined) : setSelectedNodeID(node.id);
  };
  (0, _react.useEffect)(function () {
    setSelectedNodeID(nodes[0].id);
  }, []);
  (0, _react.useEffect)(function () {
    setNodes(props.initialNodes.map(function (node) {
      return _objectSpread(_objectSpread({}, node), {}, {
        data: _objectSpread(_objectSpread({}, node.data), {}, {
          selected: node.id === selectedNodeID
        })
      });
    }));
  }, [selectedNodeID, setNodes, props.initialNodes]);
  var nodeColor = function nodeColor(node) {
    var _node$data;
    return (node === null || node === void 0 ? void 0 : (_node$data = node.data) === null || _node$data === void 0 ? void 0 : _node$data.color) || "#ccc";
  };
  var selectedNode = (0, _react.useMemo)(function () {
    return nodes.find(function (node) {
      return node.id === selectedNodeID;
    });
  }, [selectedNodeID, nodes]);
  var validLink = (0, _react.useMemo)(function () {
    var _selectedNode$data, _selectedNode$data2, _selectedNode$data3;
    if (!(selectedNode !== null && selectedNode !== void 0 && (_selectedNode$data = selectedNode.data) !== null && _selectedNode$data !== void 0 && _selectedNode$data.link)) return "";
    var hasLink = selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data2 = selectedNode.data) === null || _selectedNode$data2 === void 0 ? void 0 : _selectedNode$data2.nextFiles.some(function (file) {
      return file.name.match("page") || file.name.match("route");
    });
    return hasLink ? selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data3 = selectedNode.data) === null || _selectedNode$data3 === void 0 ? void 0 : _selectedNode$data3.link : "";
  }, [selectedNode]);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].flow,
    "data-menu": selectedNodeID !== undefined ? "true" : "false"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashBoard
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashHeader
  }, /*#__PURE__*/_react["default"].createElement("p", null, "Route information"), /*#__PURE__*/_react["default"].createElement("div", {
    onClick: closePanel,
    style: {
      cursor: "pointer"
    }
  }, /*#__PURE__*/_react["default"].createElement(_icons.CloseIcon, {
    color: "#ccc"
  }))), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashGroup
  }, /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupHeader
  }, "Name:"), /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupItem
  }, selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data4 = selectedNode.data) === null || _selectedNode$data4 === void 0 ? void 0 : _selectedNode$data4.name)), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashGroup
  }, /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupHeader
  }, "Link:"), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashGroupItem
  }, /*#__PURE__*/_react["default"].createElement("span", {
    className: _flowModule["default"].sideDashGroupLink,
    dangerouslySetInnerHTML: {
      __html: validLink !== "" ? selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data5 = selectedNode.data) === null || _selectedNode$data5 === void 0 ? void 0 : _selectedNode$data5.link.replace(/\//g, "/<wbr>").replace(/\./g, ".<wbr>").replace(/\-/g, "-<wbr>") : "None"
    }
  }), /*#__PURE__*/_react["default"].createElement("a", {
    href: validLink,
    style: {
      marginLeft: "auto"
    },
    target: "_blank",
    rel: "noreferrer",
    className: _flowModule["default"].sideDashGroupLinkIcon,
    "data-valid": validLink !== "" ? "true" : "false"
  }, /*#__PURE__*/_react["default"].createElement(_icons.ExternalLinkIcon, {
    color: validLink !== "" ? "#000" : "#ccc"
  })))), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashGroup
  }, /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupHeader
  }, "Path:"), /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupItem
  }, selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data6 = selectedNode.data) === null || _selectedNode$data6 === void 0 ? void 0 : _selectedNode$data6.path)), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashGroup
  }, /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupHeader
  }, "Type:"), /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupItem
  }, selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data7 = selectedNode.data) === null || _selectedNode$data7 === void 0 ? void 0 : _selectedNode$data7.type)), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashGroup
  }, /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupHeader
  }, "NextJS files:"), selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data8 = selectedNode.data) === null || _selectedNode$data8 === void 0 ? void 0 : (_selectedNode$data8$n = _selectedNode$data8.nextFiles) === null || _selectedNode$data8$n === void 0 ? void 0 : _selectedNode$data8$n.map(function (file) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      key: file.name,
      className: _flowModule["default"].sideDashGroupItem
    }, file.name, ".", file.extension, file.isClient ? /*#__PURE__*/_react["default"].createElement("span", {
      className: _flowModule["default"].sideDashGroupItemRightTag
    }, "client") : /*#__PURE__*/_react["default"].createElement("span", {
      className: _flowModule["default"].sideDashGroupItemRightTag
    }, "RSC"));
  }), (selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data9 = selectedNode.data) === null || _selectedNode$data9 === void 0 ? void 0 : (_selectedNode$data9$n = _selectedNode$data9.nextFiles) === null || _selectedNode$data9$n === void 0 ? void 0 : _selectedNode$data9$n.length) === 0 ? /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupItem + " " + _flowModule["default"].sideDashGroupItemError
  }, "None") : null), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashGroup
  }, /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupHeader
  }, "Other files:"), selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data10 = selectedNode.data) === null || _selectedNode$data10 === void 0 ? void 0 : (_selectedNode$data10$ = _selectedNode$data10.otherFiles) === null || _selectedNode$data10$ === void 0 ? void 0 : _selectedNode$data10$.map(function (file) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      key: file.name,
      className: _flowModule["default"].sideDashGroupItem
    }, file.name, ".", file.extension, file.isClient ? /*#__PURE__*/_react["default"].createElement("span", {
      className: _flowModule["default"].sideDashGroupItemRightTag
    }, "client") : /*#__PURE__*/_react["default"].createElement("span", {
      className: _flowModule["default"].sideDashGroupItemRightTag
    }, "RSC"));
  }), (selectedNode === null || selectedNode === void 0 ? void 0 : (_selectedNode$data11 = selectedNode.data) === null || _selectedNode$data11 === void 0 ? void 0 : (_selectedNode$data11$ = _selectedNode$data11.otherFiles) === null || _selectedNode$data11$ === void 0 ? void 0 : _selectedNode$data11$.length) === 0 ? /*#__PURE__*/_react["default"].createElement("p", {
    className: _flowModule["default"].sideDashGroupItem
  }, "None") : null), /*#__PURE__*/_react["default"].createElement("div", {
    className: _flowModule["default"].sideDashFooter
  }, /*#__PURE__*/_react["default"].createElement("p", null, "v1.1.3"), /*#__PURE__*/_react["default"].createElement("a", {
    href: "https://github.com/DiiiaZoTe/next-route-visualizer"
  }, /*#__PURE__*/_react["default"].createElement("span", null, "Github"), /*#__PURE__*/_react["default"].createElement(_icons.GithubLogo, null)))), /*#__PURE__*/_react["default"].createElement(_reactflow["default"], {
    nodes: nodes,
    onNodesChange: onNodesChange,
    edges: edges,
    onNodeClick: getNodeSelected,
    onEdgesChange: onEdgesChange,
    nodeTypes: nodeTypes,
    defaultEdgeOptions: defaultEdgeOptions,
    zoomOnDoubleClick: false,
    fitView: true,
    snapToGrid: true,
    proOptions: {
      hideAttribution: true
    },
    className: _flowModule["default"].reactFlow
  }, /*#__PURE__*/_react["default"].createElement(_reactflow.Background, {
    color: "#fafafa",
    variant: _reactflow.BackgroundVariant.Lines,
    gap: [_constants.NODE_WIDTH, _constants.NODE_HEIGHT],
    lineWidth: _constants.NODE_BORDER
  }), /*#__PURE__*/_react["default"].createElement(_reactflow.MiniMap, {
    nodeColor: nodeColor,
    nodeStrokeWidth: 3,
    zoomable: true,
    pannable: true,
    position: "bottom-right"
  }), /*#__PURE__*/_react["default"].createElement(_reactflow.Panel, {
    position: "top-left",
    className: _flowModule["default"].flowPanel
  }, /*#__PURE__*/_react["default"].createElement("button", {
    type: "button",
    onClick: refreshPage,
    className: _flowModule["default"].refreshButton
  }, /*#__PURE__*/_react["default"].createElement("span", {
    className: _flowModule["default"].refreshIcon
  }, /*#__PURE__*/_react["default"].createElement(_icons.RefreshIcon, {
    color: "#000"
  }))))));
};
var _default = Flow;
exports["default"] = _default;
