// next-route-visualizer/dist/visualizer/components/flow/nodes-flow.js
"use strict";
"use client";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _reactflow = require("reactflow");
var _constants = require("../../constants");
var _icons = require("../icons");
var _nodesFlowModule = _interopRequireDefault(require("../../styles/nodes-flow.module.css"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var __jsx = _react["default"].createElement;
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var SELECTION_BOX = 50;
var HandleCommonStyle = {
  minWidth: "0px",
  minHeight: "0px",
  width: "".concat(_constants.HANDLE_SIZE, "px"),
  height: "".concat(_constants.HANDLE_SIZE, "px"),
  background: "transparent"
};
var Box = {
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-around",
  alignItems: "center",
  flexGrow: 1,
  width: "".concat(_constants.NODE_WIDTH, "px"),
  height: "".concat(_constants.NODE_HEIGHT, "px"),
  padding: "1rem",
  fontSize: "1.2rem",
  boxSizing: "border-box"
};
var titleStyle = {
  textAlign: "center",
  fontWeight: "bolder"
};
var SelectedBox = {
  position: "absolute",
  top: "-".concat(SELECTION_BOX / 2, "px"),
  left: "-".concat(SELECTION_BOX / 2, "px"),
  width: "calc(100% + ".concat(SELECTION_BOX, "px)"),
  height: "calc(100% + ".concat(SELECTION_BOX, "px)"),
  border: "".concat(_constants.NODE_BORDER * 2, "px solid #000"),
  borderRadius: "".concat(_constants.NODE_BORDER_RADIUS + 5, "px")
};
var RouteNode = function RouteNode(_ref) {
  var _data$nextFiles, _data$nextFiles2, _data$nextFiles3, _data$nextFiles4, _data$nextFiles5, _data$nextFiles6;
  var id = _ref.id,
    data = _ref.data;
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", {
    style: Box
  }, /*#__PURE__*/_react["default"].createElement("p", {
    style: _objectSpread({
      color: data.borderColor
    }, titleStyle),
    className: _nodesFlowModule["default"].truncateText
  }, data.name), /*#__PURE__*/_react["default"].createElement("div", {
    className: _nodesFlowModule["default"].nodeIconsContainer
  }, (_data$nextFiles = data.nextFiles) !== null && _data$nextFiles !== void 0 && _data$nextFiles.find(function (file) {
    return file.name.match("page");
  }) ? /*#__PURE__*/_react["default"].createElement(_icons.FilesPageIcon, {
    color: data.borderColor
  }) : null, (_data$nextFiles2 = data.nextFiles) !== null && _data$nextFiles2 !== void 0 && _data$nextFiles2.find(function (file) {
    return file.name.match("error") || file.name.match("error");
  }) ? /*#__PURE__*/_react["default"].createElement(_icons.FilesErrorIcon, {
    color: data.borderColor
  }) : null, (_data$nextFiles3 = data.nextFiles) !== null && _data$nextFiles3 !== void 0 && _data$nextFiles3.find(function (file) {
    return file.name.match("layout") || file.name.match("template");
  }) ? /*#__PURE__*/_react["default"].createElement(_icons.FilesLayoutIcon, {
    color: data.borderColor
  }) : null, (_data$nextFiles4 = data.nextFiles) !== null && _data$nextFiles4 !== void 0 && _data$nextFiles4.find(function (file) {
    return file.name.match("route");
  }) ? /*#__PURE__*/_react["default"].createElement(_icons.FilesRouteIcon, {
    color: data.borderColor
  }) : null, (_data$nextFiles5 = data.nextFiles) !== null && _data$nextFiles5 !== void 0 && _data$nextFiles5.find(function (file) {
    return file.name.match("loading");
  }) ? /*#__PURE__*/_react["default"].createElement(_icons.FilesLoadingIcon, {
    color: data.borderColor
  }) : null)), /*#__PURE__*/_react["default"].createElement(_reactflow.Handle, {
    type: "target",
    position: _reactflow.Position.Top,
    style: _objectSpread(_objectSpread({}, HandleCommonStyle), {}, {
      top: "".concat(-_constants.HANDLE_SIZE, "px")
    }),
    id: "Normal"
  }), /*#__PURE__*/_react["default"].createElement(_reactflow.Handle, {
    type: "source",
    position: _reactflow.Position.Right,
    style: _objectSpread(_objectSpread({}, HandleCommonStyle), {}, {
      right: "".concat(-_constants.HANDLE_SIZE, "px")
    }),
    id: "Group"
  }), /*#__PURE__*/_react["default"].createElement(_reactflow.Handle, {
    type: "target",
    position: _reactflow.Position.Left,
    style: _objectSpread(_objectSpread({}, HandleCommonStyle), {}, {
      left: "".concat(-_constants.HANDLE_SIZE, "px")
    }),
    id: "Group"
  }), /*#__PURE__*/_react["default"].createElement(_reactflow.Handle, {
    type: "source",
    position: _reactflow.Position.Bottom,
    style: _objectSpread(_objectSpread({}, HandleCommonStyle), {}, {
      bottom: "".concat(-_constants.HANDLE_SIZE, "px")
    }),
    id: "Normal"
  }), /*#__PURE__*/_react["default"].createElement("div", {
    style: _objectSpread(_objectSpread({}, SelectedBox), {}, {
      borderColor: data.borderColor
    }),
    className: _nodesFlowModule["default"].nodeSelected,
    "data-selected": data.selected ? "true" : "false"
  }), (_data$nextFiles6 = data.nextFiles) === null || _data$nextFiles6 === void 0 ? void 0 : _data$nextFiles6.map(function (file) {
    return (
      // if page file then show client if isClient otherwise show SSR
      file.name.startsWith("page") ? file.isClient ? /*#__PURE__*/_react["default"].createElement("p", {
        className: _nodesFlowModule["default"].nodeClientServerType,
        style: {
          background: data.borderColor
        }
      }, "CLIENT") : /*#__PURE__*/_react["default"].createElement("p", {
        className: _nodesFlowModule["default"].nodeClientServerType,
        style: {
          background: data.borderColor
        }
      }, "SERVER") : null
    );
  }));
};
var _default = /*#__PURE__*/(0, _react.memo)(RouteNode);
exports["default"] = _default;
