// next-route-visualizer/dist/visualizer/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NODE_WIDTH = exports.NODE_SPACING_Y = exports.NODE_SPACING_X = exports.NODE_SIZE = exports.NODE_HEIGHT = exports.NODE_BORDER_RADIUS = exports.NODE_BORDER = exports.NEXT_FILES = exports.HANDLE_SIZE = exports.FALLBACK_BASE_URL = exports.ALLOWED_EXTENSIONS = void 0;
var ALLOWED_EXTENSIONS = ['.tsx', '.ts', '.js', '.jsx'];
exports.ALLOWED_EXTENSIONS = ALLOWED_EXTENSIONS;
var NEXT_FILES = ['page.', 'layout.', 'template.', 'error.', 'not-found.', 'route.', 'loading.'];
exports.NEXT_FILES = NEXT_FILES;
var FALLBACK_BASE_URL = 'http://localhost:3000';
exports.FALLBACK_BASE_URL = FALLBACK_BASE_URL;
var NODE_SIZE = 160;
exports.NODE_SIZE = NODE_SIZE;
var NODE_SPACING_X = NODE_SIZE;
exports.NODE_SPACING_X = NODE_SPACING_X;
var NODE_SPACING_Y = NODE_SIZE;
exports.NODE_SPACING_Y = NODE_SPACING_Y;
var NODE_WIDTH = NODE_SIZE;
exports.NODE_WIDTH = NODE_WIDTH;
var NODE_HEIGHT = NODE_SIZE;
exports.NODE_HEIGHT = NODE_HEIGHT;
var NODE_BORDER = 2;
exports.NODE_BORDER = NODE_BORDER;
var NODE_BORDER_RADIUS = 10;
exports.NODE_BORDER_RADIUS = NODE_BORDER_RADIUS;
var HANDLE_SIZE = 5;
exports.HANDLE_SIZE = HANDLE_SIZE;
