// next-route-visualizer/dist/visualizer/visualizer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Visualizer = void 0;
var _react = _interopRequireDefault(require("react"));
var _lib = _interopRequireDefault(require("./lib"));
var _flow = _interopRequireDefault(require("./components/flow/flow"));
var __jsx = _react["default"].createElement;
var Visualizer = function Visualizer(_ref) {
  var path = _ref.path,
    _ref$baseURL = _ref.baseURL,
    baseURL = _ref$baseURL === void 0 ? "http://localhost:3000" : _ref$baseURL,
    _ref$displayColocatin = _ref.displayColocating,
    displayColocating = _ref$displayColocatin === void 0 ? false : _ref$displayColocatin;
  var _getNodesEdges = (0, _lib["default"])({
      path: path,
      baseURL: baseURL,
      displayColocating: displayColocating
    }),
    nodes = _getNodesEdges.nodes,
    edges = _getNodesEdges.edges;
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_flow["default"], {
    initialNodes: nodes,
    initialEdges: edges
  }));
};
exports.Visualizer = Visualizer;
